modules/systemd-journal/journal-reader.c

```diff
@@ -720,32 +735,37 @@ _journal_apply_matches(JournalReader *self)
   return TRUE;
 }
 
-static void
+static gboolean
 _init_persist_name(JournalReader *self)
 {
 #if SYSLOG_NG_HAVE_JOURNAL_NAMESPACES
   if (strcmp(self->options->namespace, "*") != 0)
     {
       self->persist_name = g_strdup_printf("systemd_journal(%s)", self->options->namespace);
-      return;
+      if (!_namespace_is_already_in_use(self, self->options->namespace))
+        return TRUE;
+      else
+        return FALSE;
     }
 #endif
   self->persist_name = g_strdup("systemd-journal");
+  if (!_namespace_is_already_in_use(self, self->persist_name))
```

Comment:
Let's use `*` here as the namespace.

Reply:
I have modified the call to pass `self->options->namespace` just like in the call above this one. In the case of this call, `self->options->namespace` is always expected to be "*" because of the `if` clause above.